### PR TITLE
misc: Apply changes generated by pyupgrade

### PR DESCRIPTION
**Pull request by adamantike**
Changes automatically made by
[`pyupgrade`](https://github.com/asottile/pyupgrade), using the `--py311-plus` flag, to be compatible with Python 3.11.

### CODE REVIEW

File: backend/endpoints/tests/test_identity.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def test_login_logout(admin_user):
     assert response.status_code == 405
 
     basic_auth = base64.b64encode(
-        "test_admin:test_admin_password".encode("ascii")
+        b"test_admin:test_admin_password"
```
Comment (gantoine) [inline backend/endpoints/tests/test_identity.py]: are these equivalent?

Comment (adamantike) [reply backend/endpoints/tests/test_identity.py]: Default argument for `.encode()` is `"utf-8"`. However, as this testing string doesn't include any non-UTF8 characters, it is indeed equivalent.

```python
>>> b64encode("test_admin:test_admin_password".encode("ascii")).decode()
'dGVzdF9hZG1pbjp0ZXN0X2FkbWluX3Bhc3N3b3Jk'
>>> b64encode("test_admin:test_admin_password".encode()).decode()
'dGVzdF9hZG1pbjp0ZXN0X2FkbWluX3Bhc3N3b3Jk'
>>> b64encode(b"test_admin:test_admin_password").decode()
'dGVzdF9hZG1pbjp0ZXN0X2FkbWluX3Bhc3N3b3Jk'
```

**Review by gantoine — APPROVED** [review]

**Comment by gantoine** [comment]
Thanks for all these PRs! For future ones, it would be great if you could setup commit signing, so gh can validate you're authoring these commits. https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

File: backend/endpoints/responses/oauth.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
-from typing_extensions import NotRequired, TypedDict
+from typing import NotRequired
+
+from typing_extensions import TypedDict
```
Comment (gantoine) [inline backend/endpoints/responses/oauth.py:3]: Getting formatter errors for these changes

```
 5:1   high  Module "typing" has no attribute "NotRequired"                                        mypy/attr-defined  
 5:1   note  Use `from typing_extensions import NotRequired` instead. See                          mypy/note          
               https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-new-additions-to-t                     
               he-typing-module
```

**Review by gantoine — CHANGES_REQUESTED** [review]

File: backend/endpoints/responses/oauth.py (hunk 2 above)
Comment (adamantike) [reply backend/endpoints/responses/oauth.py:3]: The formatter should be running with an older Python version, as `NotRequired` was added to `typing` [in Python 3.11](https://docs.python.org/3/library/typing.html#typing.NotRequired).

**Comment by adamantike** [comment]
> For future ones, it would be great if you could setup commit signing, so gh can validate you're authoring these commits.

All set!

File: backend/endpoints/responses/oauth.py (hunk 2 above)
Comment (adamantike) [reply backend/endpoints/responses/oauth.py:3]: Last commit updates Python version used by trunk, that should fix the CI run!

**Review by gantoine — APPROVED** [review]

**Pull request merged**